src/main/java/com/launchdarkly/openfeature/serverprovider/Provider.java

```diff
@@ -145,44 +148,74 @@ public void initialize(EvaluationContext evaluationContext) throws Exception {
         });
         // Listen for future status changes.
         client.getDataSourceStatusProvider().addStatusListener((res) -> {
-            switch (res.getState()) {
-                // We will not re-enter INITIALIZING, but it is here to make the switch exhaustive.
-                case INITIALIZING: {
-                }
-                break;
-                case INTERRUPTED: {
-                    state = ProviderState.STALE;
-                    var message = res.getLastError() != null ? res.getLastError().getMessage() : "encountered an unknown error";
-                    emitProviderStale(ProviderEventDetails.builder().message(message).build());
+            handleDataSourceStatus(res);
+        });
+
+        handleDataSourceStatus(client.getDataSourceStatusProvider().getStatus());
```

Comment:
Manually fetch data source status in case it updated before the callback was registered.